src/screens/SpeedUpAndCancelSheet.js

```diff
@@ -120,16 +129,26 @@ export default function SpeedUpAndCancelSheet() {
   const {
     params: { type, tx },
   } = useRoute();
+  const [ready, setReady] = useState(false);
   const [isKeyboardVisible, showKeyboard, hideKeyboard] = useBooleanState();
   const [minGasPrice, setMinGasPrice] = useState(
     calcMinGasPriceAllowed(tx.gasPrice)
   );
   const fetchedTx = useRef(false);
-  const [gasLimit, setGasLimit] = useState(tx.gasLimit);
+  const [gasLimit, setGasLimit] = useState(
+    tx.gasLimit &&
+      (isHexString(tx.gasLimit) ? tx.gasLimit : toHex(tx.gasLimit.toString()))
+  );
   const [to, setTo] = useState(tx.to);
-  const [data, setData] = useState(tx.data);
-  const [value, setValue] = useState(tx.value);
-  const [nonce, setNonce] = useState(tx.nonce);
+  const [data, setData] = useState(
+    tx.data && (isHexString(tx.data) ? tx.data : toHex(tx.data.toString()))
+  );
+  const [value, setValue] = useState(
+    tx.value && (isHexString(tx.value) ? tx.value : toHex(tx.value.toString()))
```

Comment:
same here with 0 issue (not being hex'd)

Reply:
same